internal/processor/filterspan/filterspan.go

```diff
@@ -125,7 +125,7 @@ func (mp *propertiesMatcher) MatchSpan(span pdata.Span, resource pdata.Resource,
 
 // serviceNameForResource gets the service name for a specified Resource.
 func serviceNameForResource(resource pdata.Resource) string {
-	if resource.IsNil() {
+	if resource.IsEmpty() {
 		return "<nil-resource>"
 	}
```

Comment:
I think this can also be deleted. The distinction between "<nil-resource>" and "<nil-service-name>" is artificial and unnecessary.

Reply:
Yes agreed. removed even the distinction.